src/effects/backends/builtin/metronomeeffect.cpp

```diff
@@ -26,40 +26,39 @@ std::size_t playMonoSamplesWithGain(std::span<const CSAMPLE> monoSource,
     return framesPlayed;
 }
 
-template<class T>
-std::span<T> subspan_clamped(std::span<T> in, typename std::span<T>::size_type offset) {
-    // TODO (Swiftb0y): should we instead create a wrapper type that implements
-    // UB-free "clamped" operations?
-    return in.subspan(std::min(offset, in.size()));
-}
-
 constexpr std::size_t framesPerBeat(
         mixxx::audio::SampleRate framesPerSecond, double beatsPerMinute) {
     double framesPerMinute = framesPerSecond * 60;
     return static_cast<std::size_t>(framesPerMinute / beatsPerMinute);
 }
-// returns where in the output buffer to start playing the next click.
-// If there the span is empty, no click should be played yet.
-std::span<CSAMPLE> syncedClickOutput(double beatFractionBufferEnd,
-        std::optional<GroupFeatureBeatLength> beatLengthAndScratch,
-        std::span<CSAMPLE> output) {
-    if (!beatLengthAndScratch.has_value() || beatLengthAndScratch->scratch_rate == 0.0) {
-        return {};
-    }
-    double beatLength = beatLengthAndScratch->frames / beatLengthAndScratch->scratch_rate;
 
+std::size_t getbeatToBufferEndSamplesSynced(
+        double beatFractionBufferEnd,
+        const GroupFeatureBeatLength& beatLengthAndScratch) {
+    if (beatLengthAndScratch.scratch_rate == 0.0) {
+        // track paused, nothing to do.
+        return 0;
+    }
```

Comment:
this is a bug. In the (frankly unlikely but still possible) case that the click should be played with 0 offset, this will swallow the click. This happens when `beatFractionBufferEnd` is 0 (or 1, depending on `needsPreviousBeat`).

Reply:
The return value is used in `std::span.last()` 0 creates a span of size 0 that does not play a sound. But I have also placed a condition that a zero sized span is not created and the call is sipped.

Reply:
Ah I see, I had it backwards.